### PR TITLE
pkg: storage: fix additional registries

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

File: pkg/storage/image.go [hunk 1]
```diff
@@ -310,13 +313,107 @@ func isValidHostname(hostname string) bool {
...
+		remoteName = reposName
+	} else {
+		remoteName, err = withName(remoteNameStr)
+	}
+	return
+}
+
+func validateName(name string) error {
+	if err := validateID(strings.TrimPrefix(name, defaultHostname+"/")); err == nil {
+		return fmt.Errorf("Invalid repository name (%s), cannot specify 64-byte hexadecimal strings", name)
```
Comment (TomSweeneyRedHat) [inline pkg/storage/image.go]: does this error only get raised if you have a 64-byte hex string?

Comment (runcom) [reply pkg/storage/image.go]: I'd say so if you follow the function call to validateID https://github.com/kubernetes-incubator/cri-o/pull/858/files#diff-4366b8905faf0bbe4a80725e573a3b64R355 ...

File: pkg/storage/image.go [hunk 2]
```diff
@@ -310,13 +313,107 @@ func isValidHostname(hostname string) bool {
...
+	}
+	return
+}
+
+// normalize returns a repository name in its normalized form, meaning it
+// will contain library/ prefix for official images.
+func normalize(name string) (string, error) {
+	host, remoteName := splitHostname(name)
+	if strings.ToLower(remoteName) != remoteName {
+		return "", errors.New("invalid reference format: repository name must be lowercase")
```
Comment (TomSweeneyRedHat) [inline pkg/storage/image.go]: can/should we just throw out an informational message and lower case it for them?

Comment (runcom) [reply pkg/storage/image.go]: No, that becomes an API then (accept uppercase repository name). People should not rely on that and we must error out

**Comment by runcom** [comment]
@mrunalp PTAL

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**